orca-queue/src/main/kotlin/com/netflix/spinnaker/orca/q/handler/RunTaskHandler.kt

```diff
@@ -168,8 +179,8 @@ class RunTaskHandler(
   private fun Task.backoffPeriod(taskModel: com.netflix.spinnaker.orca.pipeline.model.Task, stage: Stage): TemporalAmount =
     when (this) {
       is RetryableTask -> Duration.ofMillis(
-        getDynamicBackoffPeriod(stage, Duration.ofMillis(System.currentTimeMillis() - (taskModel.startTime
-          ?: 0)))
+        Math.min(getDynamicBackoffPeriod(stage, Duration.ofMillis(System.currentTimeMillis() - (taskModel.startTime
```

Comment:
Constrains the maximum backoff so that a task won't get rescheduled too far in the future where its locks won't get extended. Essentially hijacks the WaitStage's attempt to backoff until it would be ready to complete and makes sure it gets re-run at least often enough to keep the locks alive

Reply:
^ would be a great code comment, that